jena-arq/src/main/java/org/apache/jena/sparql/exec/http/QueryExecHTTP.java

```diff
@@ -441,13 +455,28 @@ private static long asMillis(long duration, TimeUnit timeUnit) {
         return (duration < 0) ? duration : timeUnit.toMillis(duration);
     }
 
+    private void raiseException(String errorMsg,
+            HttpRequest request, HttpResponse<InputStream> response, InputStream in) {
+        int bodySummaryLength = 1024;
+        int statusCode = response.statusCode();
+        String statusCodeMsg = HttpSC.getMessage(statusCode);
+        String bodyStr;
+        try {
+            bodyStr = in == null ? "(no data supplied)" : IO.abbreviateAsUTF8(in, bodySummaryLength, "...");
```

Comment:
Should use the `Content-Type` of the response for the character set and default to UTF-8. 

Especially as this might be natural language text, JSON or HTML here.

Reply:
Hm, the squashing breaks the code references... Anyway, the raiseException method now attempts to use the response charset and falls back to UTF8.